CBLClient/Apps/CBLTestServer-C/src/CollectionMethods.cpp

```diff
@@ -0,0 +1,427 @@
+            else {
+                auto collectionName = CBLCollection_Name(collection);
+                write_serialized_body(conn, FLString(collectionName));
+            }
+        });
+    }
+
+    //creating a collection in the scope
+    void collection_createCollection(json& body, mg_connection* conn) {
+        const auto scopeName = body["scopeName"].get<string>();
+        const auto collectionName = body["collectionName"].get<string>();
+        with<CBLDatabase *>(body,"database", [conn, &collectionName, &scopeName](CBLDatabase* db)
+        {
+            CBLError* err = new CBLError();
+            CBLCollection* newCollection = CBLDatabase_CreateCollection(db, flstr(collectionName),  flstr(scopeName), err);
+            if(err->code!=0)
+                write_serialized_body(conn, err->code);
+            else
+                write_serialized_body(conn, memory_map::store(newCollection, CBLCollection_EntryDelete));
+        });
+    }
+    
+    //document count for collection.
+    void collection_documentCount(json &body, mg_connection* conn) {
+        with<CBLCollection *>(body,"collectionName",[conn,&body](CBLCollection* collectionName){
+            write_serialized_body(conn, CBLCollection_Count(collectionName));
+        });
+    }
+    
+    //collection name using collection object
+    void collection_getCollectionName(json &body, mg_connection* conn) {
+        with<CBLCollection *>(body, "collection",[conn,&body](CBLCollection* collection) {
+            write_serialized_body(conn, CBLCollection_Name(collection));
+        });
+    }
+
+    //delete collection in a scope
+    void collection_deleteCollection(json& body, mg_connection* conn) {
+        const auto scopeName = body["scopeName"].get<string>();
+        const auto collectionName = body["collectionName"].get<string>();
+        with<CBLDatabase *>(body,"database",[conn, &scopeName, &collectionName](CBLDatabase* db) {
+            CBLError* err = new CBLError();
+            auto is_deleted = CBLDatabase_DeleteCollection(db, flstr(collectionName), flstr(scopeName), err);
+            if(err->code!=0)
+                write_serialized_body(conn, err->code);
+            else
+                write_serialized_body(conn, is_deleted);
+        });
+    }
+
+    //names of all collection in a scope
+    void collection_collectionNames(json& body, mg_connection* conn) {
+        const auto scopeName = body["scopeName"].get<string>();
+        with<CBLDatabase *>(body,"database",[conn,&scopeName](CBLDatabase* db) {
+            json keys(0, nullptr);
+            CBLError* err = new CBLError();
+            FLMutableArray collectionNames = CBLDatabase_CollectionNames(db, flstr(scopeName), err);
+            if(err->code!=0)
+            {
+                write_serialized_body(conn, err->code);
+            }
+            else
+            {
+                FLArrayIterator i;
+                FLArrayIterator_Begin(collectionNames, &i);
+                do {
```

Comment:
I'm always suspicious about while loops. Is there a way to do it in a for loop (or better yet, get the collection names using CBL API?)

Reply:
Api is used to get collection names. To return the names to the testkit it needs to be converted to json. This loop is doing that.
\

Reply:
Thanks, I guess I was trying to ask if there's a way to  just return what the API returns without needing to iterate over the names.

Reply:
reinterpret_cast<const FLValue>(collectionNames)

I am using this in the new commit